### PR TITLE
Update to Serde 0.9.

### PR DESCRIPTION
**Pull request by jimmycuadra**
Serde 0.9 was recently released and required a few changes to update.

### CODE REVIEW

**Comment by FlashCat** [comment]
Thanks for the pull request, and welcome! The contain-rs team is excited to review your changes, and you should hear from @huonw (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

**Comment by jimmycuadra** [comment]
Clippy doesn't seem to have been updated for the latest nightly Rust, but my changes should be okay without clippy passing.

File: src/serde.rs [hunk 1]
```diff
@@ -55,15 +57,19 @@ impl<K, V> Visitor for LinkedHashMapVisitor<K, V>
 {
     type Value = LinkedHashMap<K, V>;
 
+    fn expecting(&self, formatter: &mut Formatter) -> FmtResult {
+        write!(formatter, "linked hash map")
```
Comment (dtolnay) [inline src/serde.rs]: I would prefer to keep it simple and say `"a map"` here. This is what we do for the built-in impls - BTreeMap and HashMap.

An example of a message you get from this is:

> invalid type: string "xyz", expected linked hash map

The user is not responsible for figuring out how to represent a linked hash map in their format. They should just see:

> invalid type: string "xyz", expected a map

Comment (jimmycuadra) [inline src/serde.rs]: Good suggestion! Updated.

**Comment by apasel422** [comment]
This looks fine to me, but Travis is still failing on nightly. Some other problem with clippy?

**Comment by dtolnay** [comment]
Clippy has not been updated to the most recent nightly yet. You can follow along here: https://github.com/Manishearth/rust-clippy/pull/1505. It is not related to this PR.

**Comment by jimmycuadra** [comment]
Thanks for merging! Would someone release a new version to crates.io, please?

**Comment by apasel422** [comment]
Published on crates.io as 0.4.0.

**Pull request merged**